src/lib.rs

```diff
@@ -17,16 +17,24 @@
 #[cfg(bench)]
 extern crate test;
 
-use std::cmp::Reverse;
+mod single_random_draw;
 
 #[cfg(any(test, feature = "rand"))]
-use rand::{seq::SliceRandom, thread_rng};
+pub use crate::single_random_draw::select_coins_srd;
+
+/// The UTXO values needed to perform coin selection.
+ #[derive(Clone)]
+pub struct Utxo {
+    value: u64,
+    feerate: u64,
+    weight: u64
+}
 
-/// Trait that a UTXO struct must implement to be used as part of the coin selection
-/// algorithm.
-pub trait Utxo: Clone {
-    /// Return the value of the UTXO.
-    fn get_value(&self) -> u64;
+impl Utxo {
+    fn get_effective_value(&self) -> u64 {
+        let fee = self.weight * self.feerate;
```

Comment:
Depending on what is the unit of the fee rate this can be incorrect. Either define the unit or define a `FeeRate` struct that can do conversions.

Reply:
@Tibo-lg I'm not sure I follow you hear.  Can you give me a better example of how the feerate can be incorrect here?

Reply:
The fee rate is often given in sats/vbytes (e.g. [the esplora api](https://github.com/Blockstream/esplora/blob/master/API.md#get-fee-estimates)), but sometimes other units are used, so you need to define the unit you require the use to pass. Having sats/weight unit is quite uncommon.

Reply:
got it, thanks!